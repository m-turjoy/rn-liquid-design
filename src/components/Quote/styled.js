import styled from "styled-components";
import {
	width,
	borderRadius,
	fontFamily,
	fontSize,
	fontWeight,
	color,
	lineHeight,
} from "styled-system";
import { colors, fonts } from "../../config";

const QuoteWrapper = styled.View`
	${width};
	flexdirection: column;
	justify-content: center;
	align-items: center;
`;

const ImageWrapper = styled.View`
	${borderRadius};
	overflow: hidden;
`;

const Image = styled.Image`
	resize-mode: contain;
	width: 100%;
	height: 100%;
`;

const AuthorWrapper = styled.Text`
	${fontSize}
	${fontFamily}
${fontWeight}
${lineHeight}
${color}
  width: 80%;
	margin-top: 20px;
	margin-bottom: 10px;
	text-align: center;
`;
const PhraseWrapper = styled.Text`
	width: 100%;
	font-size: 14px;
	line-height: 17.5px;
	text-align: center;
	font-family: "${fonts.Regular}";
	color: ${colors.richBlackDefault};
`;

export { QuoteWrapper, AuthorWrapper, PhraseWrapper, ImageWrapper, Image };
