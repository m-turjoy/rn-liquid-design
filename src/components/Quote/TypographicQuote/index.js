import React from "react";
import { bool, number, string } from "prop-types";
import { fonts, colors } from "../../../config";
import { QuoteWrapper, PhraseWrapper, AuthorWrapper } from "./styled";

const TypographicQuote = ({
	author,
	big,
	small,
	quotation,
	width,
	fontFamily,
	fontSize,
	fontWeight,
	lineHeight,
	color,
	authorFontFamily,
	authorFontSize,
	authorFontWeight,
	authorLineHeight,
	authorColor,
}) => (
	<QuoteWrapper width={width}>
		<PhraseWrapper
			fontFamily={fontFamily || fonts.Black}
			fontSize={big ? 32 : small ? 22 : fontSize}
			fontWeight={fontWeight || null}
			lineHeight={big ? 40 : small ? 27.5 : lineHeight}
			color={color || colors.richBlackDefault}
		>
			{`»${quotation}«`}
		</PhraseWrapper>
		<AuthorWrapper
			fontFamily={authorFontFamily || fonts.Regular}
			fontSize={authorFontSize || 12}
			fontWeight={authorFontWeight || null}
			lineHeight={authorLineHeight || 15}
			color={authorColor || colors.vibrantCyanDefault}
		>
			{`– ${author}`}
		</AuthorWrapper>
	</QuoteWrapper>
);

TypographicQuote.propTypes = {
	quotation: string.isRequired,
	author: string.isRequired,
	big: bool,
	small: bool,
	width: number,
	fontFamily: string,
	fontSize: number,
	fontWeight: number,
	lineHeight: number,
	color: string,
	authorFontFamily: string,
	authorFontSize: number,
	authorFontWeight: number,
	authorLineHeight: number,
	authorColor: string,
};

TypographicQuote.defaultProps = {
	big: false,
	width: 250,
};

export default TypographicQuote;
