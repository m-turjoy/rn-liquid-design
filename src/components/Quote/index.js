import React from "react";
import { bool, node, number, string, PropTypes } from "prop-types";
import { ThemeProvider } from "styled-components";
import {
	QuoteWrapper,
	PhraseWrapper,
	AuthorWrapper,
	ImageWrapper,
	Image,
} from "./styled";
import { fonts, colors } from "../../config";
import TypographicQuote from "./TypographicQuote";
import { defaultThemeName, getThemeObject } from "../../config/theme";

const Quote = ({
	author,
	big,
	small,
	imagePath,
	quotation,
	borderRadius,
	width,
	fontFamily,
	fontSize,
	fontWeight,
	lineHeight,
	color,
	authorFontFamily,
	authorFontWeight,
	authorFontSize,
	authorColor,
	authorLineHeight,
	themeName,
}) => {
	const imgWrapperSize = (80 * width) / 100;
	let themeObj = getThemeObject(themeName);
	let themeColor = themeObj.colors.primary.base;
	const primaryColor = authorColor || themeColor;

	return (
		<ThemeProvider theme={themeObj}>
			{imagePath ? (
				<QuoteWrapper width={width}>
					<ImageWrapper
						borderRadius={borderRadius}
						height={imgWrapperSize}
						width={imgWrapperSize}
					>
						<Image source={imagePath} />
					</ImageWrapper>
					<AuthorWrapper
						fontFamily={authorFontFamily || fonts.Black}
						fontWeight={authorFontWeight || null}
						fontSize={authorFontSize || 18}
						lineHeight={authorLineHeight || 22.5}
						color={authorColor || colors.richBlackDefault}
					>
						{author}
					</AuthorWrapper>
					<PhraseWrapper
						fontFamily={fontFamily || fonts.Regular}
						fontWeight={fontWeight || null}
						fontSize={fontSize || 14}
						lineHeight={lineHeight || 17.5}
						color={color || colors.richBlackDefault}
					>
						{quotation}
					</PhraseWrapper>
				</QuoteWrapper>
			) : (
				<TypographicQuote
					width={width}
					author={author}
					big={big}
					small={small}
					quotation={quotation}
					fontFamily={fontFamily}
					fontWeight={fontWeight}
					fontSize={fontSize}
					lineHeight={lineHeight}
					color={color}
					authorColor={primaryColor}
					authorFontFamily={authorFontFamily}
					authorFontWeight={authorFontWeight}
					authorFontSize={authorFontSize}
				/>
			)}
		</ThemeProvider>
	);
};

Quote.propTypes = {
	quotation: string.isRequired,
	author: string.isRequired,
	big: bool,
	small: bool,
	/** if provided, will display a differently styled quote */
	imagePath: node,
	borderRadius: number,
	width: number,
	fontFamily: string,
	fontSize: number,
	fontWeight: number,
	color: string,
	lineHeight: string,
	authorFontFamily: string,
	authorFontWeight: string,
	authorFontSize: number,
	authorLineHeight: number,
	authorColor: string,
	themeName: PropTypes.oneOfType([
		string,
		PropTypes.shape({
			primary: PropTypes.shape({
				lightest: string,
				light: string,
				base: string,
				dark: string,
				darker: string,
			}).isRequired,
			secondary: PropTypes.shape({
				lightest: string,
				light: string,
				base: string,
				dark: string,
				darker: string,
			}).isRequired,
		}),
	]),
};

Quote.defaultProps = {
	big: false,
	small: true,
	imagePath: null,
	borderRadius: 100,
	width: 250,
	themeName: defaultThemeName,
};

export default Quote;
