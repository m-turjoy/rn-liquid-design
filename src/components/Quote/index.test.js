import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Quote from "./index";
import imageTest from "../../assets/circle.png";

configure({ adapter: new Adapter() });
describe("Quote component", () => {
	describe("Renders correctly", () => {
		test("it renders default Quote", () => {
			const tree = shallow(
				<Quote quotation="Test" author="Test" imagePath={imageTest} />
			);
			expect(tree).toMatchSnapshot();
		});
	});
});
