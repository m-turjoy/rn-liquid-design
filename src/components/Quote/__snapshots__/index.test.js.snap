// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component Renders correctly it renders default Quote 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quote
    author="Test"
    big={false}
    borderRadius={100}
    imagePath={
      Object {
        "testUri": "../../assets/circle.png",
      }
    }
    quotation="Test"
    small={true}
    themeName="vibrantCyan"
    width={250}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(View)
        width={250}
      >
        <Styled(View)
          borderRadius={100}
          height={200}
          width={200}
        >
          <Styled(Image)
            source={
              Object {
                "testUri": "../../assets/circle.png",
              }
            }
          />
        </Styled(View)>
        <Styled(Text)
          color="#1b1b25"
          fontFamily="Lato-Black"
          fontSize={18}
          fontWeight={null}
          lineHeight={22.5}
        >
          Test
        </Styled(Text)>
        <Styled(Text)
          color="#1b1b25"
          fontFamily="Lato-Regular"
          fontSize={14}
          fontWeight={null}
          lineHeight={17.5}
        >
          Test
        </Styled(Text)>
      </Styled(View)>,
      "theme": Object {
        "colors": Object {
          "black": Object {
            "base": undefined,
          },
          "primary": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "light": "#57cbd7",
            "lightest": "#abe5eb",
          },
          "richBlack": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "richBlue": Object {
            "base": "#0f69af",
            "dark": "#0e5f9e",
            "darker": "#0b4a7b",
            "darkest": "#083558",
            "light": "#3f87bf",
            "lighter": "#6fa5cf",
            "lightest": "#9fc3df",
          },
          "richGreen": Object {
            "base": "#149b5f",
            "dark": "#0d6c42",
            "darker": "#0c5d39",
            "darkest": "#094d2f",
            "light": "#5ab98f",
            "lighter": "#72c39f",
            "lightest": "#89cdaf",
          },
          "richPurple": Object {
            "base": "#503291",
            "dark": "#482d83",
            "darker": "#382366",
            "darkest": "#281949",
            "light": "#735ba7",
            "lighter": "#9684bd",
            "lightest": "#b9add3",
          },
          "richRed": Object {
            "base": "#e61e50",
            "dark": "#cf1b48",
            "darker": "#a11538",
            "darkest": "#730f28",
            "light": "#eb4b73",
            "lighter": "#f07896",
            "lightest": "#f5a5b9",
          },
          "secondary": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "light": "#ffd35b",
            "lightest": "#ffe9ad",
          },
          "sensitiveGrey": Object {
            "base": "#f8f8fc",
            "dark": "#f3f3f7",
            "darker": "#efeff1",
            "darkest": "#e9e9ed",
            "light": "#f8f8fc",
            "lighter": "#ffffff",
            "lightest": "#ffffff",
          },
          "vibrantCyan": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "darkest": "#175f67",
            "light": "#57cbd7",
            "lighter": "#81d8e1",
            "lightest": "#abe5eb",
          },
          "vibrantGreen": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "vibrantMagenta": Object {
            "base": "#eb3c96",
            "dark": "#d43687",
            "darker": "#a52a69",
            "darkest": "#761e4b",
            "light": "#ef63ab",
            "lighter": "#f38ac0",
            "lightest": "#f7b1d5",
          },
          "vibrantYellow": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "darkest": "#806419",
            "light": "#ffd35b",
            "lighter": "#ffde84",
            "lightest": "#ffe9ad",
          },
          "white": Object {
            "base": "#ffffff",
          },
        },
        "fontSizes": Array [
          8,
          10,
          12,
          14,
          16,
          18,
          20,
          24,
          36,
        ],
        "fontWeights": Array [
          100,
          200,
          300,
          400,
          500,
          600,
          700,
          800,
          900,
        ],
        "fonts": Object {
          "Black": "Lato-Black",
          "Bold": "Lato-Bold",
          "IcoMoon": "icomoon",
          "Merck": "MerckWeb-Regular",
          "Regular": "Lato-Regular",
        },
        "heights": Object {
          "Big": 50,
          "Default": 40,
        },
        "shadows": Object {
          "active": Object {
            "elevation": 40,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
          "default": Object {},
          "stacked": Object {
            "elevation": 25,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
        },
        "space": Array [
          0,
          1,
          2,
          5,
          10,
          15,
          20,
          24,
          50,
          8,
          7,
          4,
          9,
          6,
        ],
        "stacked": Object {
          "stackedFirst": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "position": "absolute",
            "zIndex": 10,
          },
          "stackedSecond": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "zIndex": 9,
          },
          "stackedThird": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 6,
            "top": 6,
            "zIndex": 8,
          },
        },
        "widths": Array [
          50,
          100,
          130,
        ],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(View)
            borderRadius={100}
            height={200}
            width={200}
          >
            <Styled(Image)
              source={
                Object {
                  "testUri": "../../assets/circle.png",
                }
              }
            />
          </Styled(View)>,
          <Styled(Text)
            color="#1b1b25"
            fontFamily="Lato-Black"
            fontSize={18}
            fontWeight={null}
            lineHeight={22.5}
          >
            Test
          </Styled(Text)>,
          <Styled(Text)
            color="#1b1b25"
            fontFamily="Lato-Regular"
            fontSize={14}
            fontWeight={null}
            lineHeight={17.5}
          >
            Test
          </Styled(Text)>,
        ],
        "width": 250,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "borderRadius": 100,
            "children": <Styled(Image)
              source={
                Object {
                  "testUri": "../../assets/circle.png",
                }
              }
            />,
            "height": 200,
            "width": 200,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "testUri": "../../assets/circle.png",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test",
            "color": "#1b1b25",
            "fontFamily": "Lato-Black",
            "fontSize": 18,
            "fontWeight": null,
            "lineHeight": 22.5,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test",
            "color": "#1b1b25",
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "fontWeight": null,
            "lineHeight": 17.5,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(View)
          width={250}
        >
          <Styled(View)
            borderRadius={100}
            height={200}
            width={200}
          >
            <Styled(Image)
              source={
                Object {
                  "testUri": "../../assets/circle.png",
                }
              }
            />
          </Styled(View)>
          <Styled(Text)
            color="#1b1b25"
            fontFamily="Lato-Black"
            fontSize={18}
            fontWeight={null}
            lineHeight={22.5}
          >
            Test
          </Styled(Text)>
          <Styled(Text)
            color="#1b1b25"
            fontFamily="Lato-Regular"
            fontSize={14}
            fontWeight={null}
            lineHeight={17.5}
          >
            Test
          </Styled(Text)>
        </Styled(View)>,
        "theme": Object {
          "colors": Object {
            "black": Object {
              "base": undefined,
            },
            "primary": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "light": "#57cbd7",
              "lightest": "#abe5eb",
            },
            "richBlack": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "richBlue": Object {
              "base": "#0f69af",
              "dark": "#0e5f9e",
              "darker": "#0b4a7b",
              "darkest": "#083558",
              "light": "#3f87bf",
              "lighter": "#6fa5cf",
              "lightest": "#9fc3df",
            },
            "richGreen": Object {
              "base": "#149b5f",
              "dark": "#0d6c42",
              "darker": "#0c5d39",
              "darkest": "#094d2f",
              "light": "#5ab98f",
              "lighter": "#72c39f",
              "lightest": "#89cdaf",
            },
            "richPurple": Object {
              "base": "#503291",
              "dark": "#482d83",
              "darker": "#382366",
              "darkest": "#281949",
              "light": "#735ba7",
              "lighter": "#9684bd",
              "lightest": "#b9add3",
            },
            "richRed": Object {
              "base": "#e61e50",
              "dark": "#cf1b48",
              "darker": "#a11538",
              "darkest": "#730f28",
              "light": "#eb4b73",
              "lighter": "#f07896",
              "lightest": "#f5a5b9",
            },
            "secondary": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "light": "#ffd35b",
              "lightest": "#ffe9ad",
            },
            "sensitiveGrey": Object {
              "base": "#f8f8fc",
              "dark": "#f3f3f7",
              "darker": "#efeff1",
              "darkest": "#e9e9ed",
              "light": "#f8f8fc",
              "lighter": "#ffffff",
              "lightest": "#ffffff",
            },
            "vibrantCyan": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "darkest": "#175f67",
              "light": "#57cbd7",
              "lighter": "#81d8e1",
              "lightest": "#abe5eb",
            },
            "vibrantGreen": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "vibrantMagenta": Object {
              "base": "#eb3c96",
              "dark": "#d43687",
              "darker": "#a52a69",
              "darkest": "#761e4b",
              "light": "#ef63ab",
              "lighter": "#f38ac0",
              "lightest": "#f7b1d5",
            },
            "vibrantYellow": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "darkest": "#806419",
              "light": "#ffd35b",
              "lighter": "#ffde84",
              "lightest": "#ffe9ad",
            },
            "white": Object {
              "base": "#ffffff",
            },
          },
          "fontSizes": Array [
            8,
            10,
            12,
            14,
            16,
            18,
            20,
            24,
            36,
          ],
          "fontWeights": Array [
            100,
            200,
            300,
            400,
            500,
            600,
            700,
            800,
            900,
          ],
          "fonts": Object {
            "Black": "Lato-Black",
            "Bold": "Lato-Bold",
            "IcoMoon": "icomoon",
            "Merck": "MerckWeb-Regular",
            "Regular": "Lato-Regular",
          },
          "heights": Object {
            "Big": 50,
            "Default": 40,
          },
          "shadows": Object {
            "active": Object {
              "elevation": 40,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
            "default": Object {},
            "stacked": Object {
              "elevation": 25,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
          },
          "space": Array [
            0,
            1,
            2,
            5,
            10,
            15,
            20,
            24,
            50,
            8,
            7,
            4,
            9,
            6,
          ],
          "stacked": Object {
            "stackedFirst": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "position": "absolute",
              "zIndex": 10,
            },
            "stackedSecond": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 3,
              "position": "absolute",
              "top": 3,
              "zIndex": 9,
            },
            "stackedThird": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 6,
              "top": 6,
              "zIndex": 8,
            },
          },
          "widths": Array [
            50,
            100,
            130,
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(View)
              borderRadius={100}
              height={200}
              width={200}
            >
              <Styled(Image)
                source={
                  Object {
                    "testUri": "../../assets/circle.png",
                  }
                }
              />
            </Styled(View)>,
            <Styled(Text)
              color="#1b1b25"
              fontFamily="Lato-Black"
              fontSize={18}
              fontWeight={null}
              lineHeight={22.5}
            >
              Test
            </Styled(Text)>,
            <Styled(Text)
              color="#1b1b25"
              fontFamily="Lato-Regular"
              fontSize={14}
              fontWeight={null}
              lineHeight={17.5}
            >
              Test
            </Styled(Text)>,
          ],
          "width": 250,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "borderRadius": 100,
              "children": <Styled(Image)
                source={
                  Object {
                    "testUri": "../../assets/circle.png",
                  }
                }
              />,
              "height": 200,
              "width": 200,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": Object {
                  "testUri": "../../assets/circle.png",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test",
              "color": "#1b1b25",
              "fontFamily": "Lato-Black",
              "fontSize": 18,
              "fontWeight": null,
              "lineHeight": 22.5,
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test",
              "color": "#1b1b25",
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
              "fontWeight": null,
              "lineHeight": 17.5,
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
